### PR TITLE
Add Ekoplaza crawler

### PR DESCRIPTION
**Pull request by annaproxy**
Support for ekoplaza.nl

### CODE REVIEW

**Comment by Tweska** [comment]
Oh nice addition, I haven't tested it yet but I noticed you didn't add the copyright notice to `argostime/crawler/ekoplaza.py`. You should do so to please @m-rtijn ... :laughing: 

Bit off topic but: how did you find this project?

**Comment by annaproxy** [comment]
Thanks, I updated the header.

Me and a friend found it from [this reddit comment](https://old.reddit.com/r/thenetherlands/comments/umdsgu/een_prijsvergelijkingswebsite_voor_supermarkten/i82bckh/) :)

**Comment by Tweska** [comment]
Seems like my promotion comment worked.

File: argostime/crawler/ekoplaza.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Argostimè. If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+import json
+import logging
+from pprint import pprint
```
Comment (m-rtijn) [inline argostime/crawler/ekoplaza.py:28]: It seems like this import isn't actually used

File: argostime/crawler/ekoplaza.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+from argostime.exceptions import PageNotFoundException
+from argostime.crawler.crawl_utils import CrawlResult
+
+
+def crawl_ekoplaza(url: str):  # -> CrawlResult:
+    """Ekoplaza crawler"""
+
+    info = url.split('product/')[-1]
+    response = requests.get(
+        f'https://www.ekoplaza.nl/api/aspos/products/url/{info}')
```
Comment (m-rtijn) [inline argostime/crawler/ekoplaza.py:43]: Interessant dat er gewoon een API achter zit, maakt het ons wel een stukje makkelijker :)

**Review by m-rtijn — COMMENTED** [review]
Thanks for the PR! I haven't had the time yet to actually test the code, but here are my first thoughts

File: argostime/crawler/ekoplaza.py (hunk 2 above)
Comment (annaproxy) [reply argostime/crawler/ekoplaza.py:42]: Er zit ook veel te veel info in die API, bijv. 'PurchasePrice' wat de inkoopprijs voor de supermarkt lijkt te zijn

File: argostime/crawler/ekoplaza.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    result.url = url
+    try:
+        result.product_name = product['Description'].title()
+    except KeyError as exception:
+        logging.error("No product name found in %s", product)
+        raise CrawlerException from exception
+
+    try:
+        result.product_code = product['DefaultScanCode']['Code']  # EAN ?
```
Comment (Tweska) [inline argostime/crawler/ekoplaza.py:64]: I checked a few products and these seem to be valid EAN codes for the products.

File: argostime/crawler/ekoplaza.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+
+import requests
+
+from argostime.exceptions import CrawlerException
+from argostime.exceptions import PageNotFoundException
+from argostime.crawler.crawl_utils import CrawlResult
+
+
+def crawl_ekoplaza(url: str):  # -> CrawlResult:
```
Comment (Tweska) [inline argostime/crawler/ekoplaza.py:37]: You should not comment out the return type.

File: argostime/crawler/ekoplaza.py (hunk 2 above)
Comment (m-rtijn) [reply argostime/crawler/ekoplaza.py:42]: Hah, wauw

File: argostime/crawler/ekoplaza.py (hunk 3 above)
Comment (Tweska) [reply argostime/crawler/ekoplaza.py:64]: So we should use EAN for both then right? So we can use that EAN in the future to match the same products across multiple shops.

Comment (m-rtijn) [reply argostime/crawler/ekoplaza.py:64]: Yes exactly

**Review by Tweska — APPROVED** [review]

**Review by m-rtijn — APPROVED** [review]

**Comment by m-rtijn** [comment]
@annaproxy Thanks a lot for adding this crawler! The newest version will be deployed to production once the CI checks are all OK.

**Pull request merged**